main.tf

```diff
@@ -32,14 +32,27 @@ resource "google_folder" "folders" {
 # https://cloud.google.com/resource-manager/docs/access-control-folders#granting_folder-specific_roles_to_enable_project_creation
 
 resource "google_folder_iam_binding" "owners" {
-  count  = "${var.set_roles ? length(var.names) * length(var.folder_admin_roles) : 0}"
-  folder = "${element(google_folder.folders.*.name, count.index / length(var.folder_admin_roles))}"
-  role   = "${element(var.folder_admin_roles, count.index % length(var.folder_admin_roles))}"
+  count = var.set_roles ? length(var.names) * length(var.folder_admin_roles) : 0
+  folder = element(
+    google_folder.folders.*.name,
+    floor(count.index / length(var.folder_admin_roles)),
```

Comment:
What was the motivation for adding floor?

Reply:
Without it, we have an error because of the floating result of the operation `count.index / length(var.folder_admin_roles)` in terraform-0.12

Reply:
@aaron-lane Without floor it simply fails on TF0.12. I've checked it